lucet-runtime/lucet-runtime-internals/src/instance.rs

```diff
@@ -1471,6 +1471,29 @@ impl TerminationDetails {
             _ => None,
         }
     }
+    /// Try to interpret the termination details as a provided exit code.
+    ///
+    /// The most consistent form of `TerminationDetails::Provided` comes from Lucet's
+    /// implementation of `proc_exit`, which exits with a `Provided` holding the given exit code.
+    /// For cases where a Lucet user simply wants "`proc_exit` or continue panicking" behavior,
+    /// `as_exitcode` can simplify handling `TerminationDetails`.
+    pub fn as_exitcode(&self) -> Option<u32> {
```

Comment:
the return type here is kinda awkward. i'd _like_ to be able to write `lucet_wasi::types::Exitcode` here, but that would make a circular dependency with `lucet_runtime`.

Reply:
its fine to use a u32 here, in my opinion. We don't want to take a dep on lucet-wasi (or wasi-common, where that type really comes from) in the runtime.

Reply:
(it is slightly unfortunate that we have to take a dep on wiggle in the runtime, but I think it is fine in this case)